modules/utils/tests/string_utils_tests.cpp

```diff
@@ -2,66 +2,135 @@
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "to",
+                                                    .end_token = ".txt",
+                                                    .include_end_token = false,
+                                                    .expected = "to/some/file" },
+                                          TestCase{ .description = "Truncate invalid tokens",
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "aaa",
+                                                    .end_token = "bbb",
+                                                    .include_end_token = true,
+                                                    .expected = "/path/to/some/file.txt" },
+                                          TestCase{ .description = "Truncate start invalid",
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "aaa",
+                                                    .end_token = ".txt",
+                                                    .include_end_token = false,
+                                                    .expected = "/path/to/some/file" },
+                                          TestCase{ .description = "Truncate end invalid",
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "some",
+                                                    .end_token = "bbb",
+                                                    .include_end_token = true,
+                                                    .expected = "some/file.txt" },
+                                          TestCase{ .description = "Truncate start and end empty",
+                                                    .str = "/path/to/some/file.txt",
+                                                    .start_token = "",
+                                                    .end_token = "",
+                                                    .include_end_token = true,
+                                                    .expected = "/path/to/some/file.txt" } };
+
+  for (const auto& test_case : TEST_CASES) {
     const auto truncated =
         truncate(test_case.str, test_case.start_token, test_case.end_token, test_case.include_end_token);
     EXPECT_EQ(truncated, test_case.expected) << test_case.description;
   }
+
+  constexpr auto CONSTEXPR_TEST_CASE = TEST_CASES[0];
+  constexpr auto CONSTEXPR_TRUNCATED =
+      truncate(CONSTEXPR_TEST_CASE.str, CONSTEXPR_TEST_CASE.start_token, CONSTEXPR_TEST_CASE.end_token,
+               CONSTEXPR_TEST_CASE.include_end_token);
+  static_assert(CONSTEXPR_TRUNCATED == CONSTEXPR_TEST_CASE.expected);
```

Comment:
actually, can you have the entire previous for loop as constexpr?
```
for (constexpr auto& test_case : TEST_CASES) {
    constexpr auto truncated =
        truncate(test_case.str, test_case.start_token, test_case.end_token, test_case.include_end_token);
    static_assert(truncated == test_case.expected);
  }
```

Reply:
A for loop can't be `constexpr`. Unless we write our own version of a for loop templated on the iteration count, not possible AFAIK.